xxxx-ewallet-blockchain-integration.md

```diff
@@ -0,0 +1,315 @@
+---
+oip: "xxxx"
+title: eWallet Blockchain Integration
+status: Active
+type: eWallet
+author: Thibault Denizet <thibault@omise.co>
+created: 2018-08-08
+---
+
+# OIP-0004: eWallet Blockchain Integration
+
+## Abstract
+
+This OIP describes the changes required to plug an eWallet onto Ethereum and Plasma.
+
+## Specification
+
+### eWallet Ethereum Name Service
+
+The eWallet Ethereum Name Service is a smart contract running on the Ethereum blockchain that allows eWallet providers to register their eWallet as being part of the network. It is a simple key/value storage where the key is the eWallet primary public key and the value their current domain name.
+
+- Option #1: Using [ENS](https://ens.domains/)
+- Option #2: Build a simple eWallet Name Service.
+
+### Blockchain Wallets
+
+Blockchain wallets will be the representations of Ethereum accounts in the eWallet. Two types of blockchain wallets will exist: `hot` and `cold`.
+
+The eWallet saves the encrypted private keys of the `hot` wallets in the database, with the encryption key being stored in a file to which only the application has access. No private keys are stored with `cold` wallets and the administrators are responsible for managing them directly. Interaction with `cold` wallets will require manual action through [Metamask](https://metamask.io/) or similar tools.
```

Comment:
From team discussion:

Env var means it's available to all process running in the container. Using a file instead means you can restrict which user and process can access it but it won't be 12-factor.

Reply:
Which means we should recommend the use of a file but allow the ENV variable, in case someone would like to deploy on Heroku for example.